lib/email/receiver.rb

```diff
@@ -422,6 +426,20 @@ def can_invite?(topic, user)
       !topic.topic_allowed_groups.where("group_id IN (SELECT group_id FROM group_users WHERE user_id = ?)", user.id).exists?
     end
 
+    private
+
+    def check_reply_to_auto_generated_header
```

Comment:
why can't you just check `@mail.header["X-Discourse-Reply-to-Auto-Generated"] == "marked"`?